bag-baggy

imans777 opened the pull request:
:|
same pull request as the previous one

fardabara committed:
961221 added bag

fardabara committed:
delete asstes pictures

fardabara committed:
961223 commited after pull

imans777 committed:
fix changes from server

fardabara committed:
961223 _app component create and count added

imans777 committed:
Merge branch 'app-bag' of https://github.com/PersianMode/app into cart-service

# Conflicts:
#	src/pages/inbox/inbox.ts

remove orderline from server
load orderlines in the beginning

imans777 committed:
Merge branch 'app-bag' of https://github.com/PersianMode/app into cart-service

# Conflicts:
#	src/pages/inbox/inbox.ts

remove orderline from server
load orderlines in the beginning

imans777 committed:
Merge branch 'app-bag' of https://github.com/PersianMode/app into cart-service

# Conflicts:
#	src/pages/inbox/inbox.ts

remove orderline from server
load orderlines in the beginning

imans777 committed:
use the user from server, instead of us sending customer_id
fix a bug in showing badge for the bag tab
send product_id along with other information when someone clicked on buy in product-view
fix a bug related to app's crashing
get products from server instead of mock
reformat products from server to match with the client
change in product-sliding (also there must be another thing done for this)

imans777 committed:
Merge remote-tracking branch 'origin/_new_app_bag' into cart-service

# Conflicts:
#	src/app/app.module.ts
#	src/pages/bag/bag.html
#	src/pages/bag/bag.ts
#	src/pages/bag/product-sliding/product-sliding.html
#	src/pages/bag/product-sliding/product-sliding.ts
#	src/pages/bag/select-count/select-count.html
#	src/pages/bag/select-count/select-count.scss
#	src/pages/bag/select-count/select-count.ts
#	src/pages/products/select-size/select-size.html

imans777 committed:
:|
reset hard to previous commits then add changes manually

ARHariri commented on src/app/app.module.ts:
```diff
@@ -73,7 +77,9 @@ import {CartService} from "../services/cart.service";
-    FilterPage
+    FilterPage,
+    ProductSliding,
+    SelectCount,
```
Please remove the components from here (Used as component not page)

ARHariri commented on src/pages/bag/bag.html:
```diff
@@ -5,6 +5,21 @@
+        <!--<button ion-button color="danger" (click)="removeThisProduct(product)">حذف<ion-icon  name="trash"></ion-icon></button>-->
+      <!--</ion-item-options>-->
+    <!--</ion-item-sliding>-->
+  <!--</ion-list>-->
```
Remove the commented lines unless need them

ARHariri commented on src/pages/bag/bag.ts:
```diff
@@ -1,33 +1,64 @@
 })
 export class BagPage {
-  products: Product[] = [];
+  products: any[] = [{
```
Use Product instances if doesn't make any error

ARHariri commented on src/pages/bag/bag.ts:
```diff
@@ -1,33 +1,64 @@
+    thumbnail: 'assets/product-pic/10.jpg',
+    color: 'سفید زرد آبی',
+    size: 'M 14 / W 15.5',
+  }];
```
Please remove the mock data and use server to fetch bag page details

ARHariri commented on src/pages/bag/product-sliding/product-sliding.html:
```diff
@@ -0,0 +1,24 @@
+        <img src="{{product.thumbnail}}" alt="">
+      </ion-thumbnail>
+      <div>
+        <p style="color:#000;">{{product.name}}</p>
```
Using class instead of declaring inline style is best practice because the style is used in other lines

ARHariri commented on src/pages/bag/product-sliding/product-sliding.ts:
```diff
@@ -0,0 +1,46 @@
+
+  actionCount() {
+    let overCtrl = this.popoverCtrl.create(SelectCount, {
+      count: 10
```
the data passed to SelectCount component must be depended on product instance count (cannot hard-coded and fix it to 10)

ARHariri commented on src/pages/bag/select-count/select-count.ts:
```diff
@@ -0,0 +1,23 @@
+    for(let i = 1; i <= _count; i++) {
+      let _countObj = {
+        key: i,
+        value: i
```
What is different between key and value when both have i value,
The value property should have the number in Persian format

ARHariri commented on src/pages/bag/select-count/select-count.ts:
```diff
@@ -0,0 +1,23 @@
+
+  constructor(public navParams: NavParams) {
+    const _count = this.navParams.get('count');
+    let _countArr = [];
```
You can use the count variable instead of using _countArr variable

ARHariri commented on src/pages/collection/collection.ts:
```diff
@@ -17,33 +17,6 @@ export class CollectionPage implements OnInit {
-  //       }]
-  //     }
-  //   }]
-  // }];
```
Remove the commented code

ARHariri commented on src/pages/inbox/inbox.ts:
```diff
@@ -36,7 +36,7 @@ export class InboxPage {
   goToProductView() {
     this.navCtrl.push(ProductViewPage, {
-      productId: '5a9eb0344f31b61a47d419b7'
+      productId: "5a9fc897bcf5d7654e7a410e"
```
Please remove the hard-coded id (Fetch product from server)

ARHariri commented on src/pages/products/select-size/select-size.ts:
```diff
@@ -29,15 +31,15 @@ export class SelectSizePage {
           return this.presentToast("لطفا مجددا تلاش کنید");
         }
 
         this.loading.dismiss();
```
If using call back function this code is working well? (because using this keyword)
Highly recommended to using promise instead of using call back function

ARHariri commented on src/pages/tabs/tabs.ts:
```diff
@@ -32,4 +32,4 @@ export class TabsPage {
   }
-}
\ No newline at end of file
+}
```
On ngOnInit function: you get cartItems and then assign it to cartNum (Should get res.length)

ARHariri commented on src/services/cart.service.ts:
```diff
@@ -1,40 +1,88 @@
+    );
+  }
+
+  addOrderline(product_id, product_instance_id, number, cb) {
```
Highly recommended to using promise and return promise instead of passing callback function

ARHariri commented on src/services/cart.service.ts:
```diff
@@ -1,40 +1,88 @@
+  loadOrderlines() {
+    this.httpService.post(`cart/items`, {data: {}}).subscribe(
+      data => {
+        console.log(data);
```
Prevent to logging anything on console in production code

ARHariri commented on src/services/cart.service.ts:
```diff
@@ -1,40 +1,88 @@
+        this.dataArray = data;
+        this.cartItems.next(this.getTotalNumber());
+      }, err => {
+        console.log("err", err);
```
Use console.error instead of console.log (For logging error, please use the error function)

ARHariri commented on src/services/cart.service.ts:
```diff
@@ -1,40 +1,88 @@
+          quantity: number,
+        });
+        this.cartItems.next(this.getTotalNumber());
+        if (!(res.n > 0 || res.nModified > 0))
```
Highly recommended to change this condition to:
res.n <= 0 && res.nModified <= 0

ARHariri commented on src/services/cart.service.ts:
```diff
@@ -1,40 +1,88 @@
+
+        if (cb) cb(null);
+      }, err => {
+        console.log('error in adding orderline', err);
```
same as error instead log function

ARHariri commented on src/services/cart.service.ts:
```diff
@@ -1,40 +1,88 @@
+    );
+  }
+
+  removeOrderline(product_instance_id, number, cb) {
```
Same as above: Highly recommended to use promise and return is instead of using callback fucntion

ARHariri commented on src/services/cart.service.ts:
```diff
@@ -1,40 +1,88 @@
+    let data = {product_instance_id, number};
+    this.httpService.post(`order/delete`, data).subscribe(
+      res => {
+        this.dataArray = this.dataArray.filter(data => data['instance_id'] === product_instance_id && number-- > 0);
```
The condition used in filter function is incorrect

ARHariri commented on src/services/cart.service.ts:
```diff
@@ -1,40 +1,88 @@
+      res => {
+        this.dataArray = this.dataArray.filter(data => data['instance_id'] === product_instance_id && number-- > 0);
+        this.cartItems.next(this.dataArray.length);
+        if (!(res.n > 0 || res.nModified > 0))
```
use: res.n <= 0 && res.nModified <= 0

ARHariri commented on src/services/cart.service.ts:
```diff
@@ -1,40 +1,88 @@
+
+        if (cb) cb(null);
+      }, err => {
+        console.log("error in removing orderline", err);
```
Same as before: using error function instead of log function

ARHariri commented on src/services/cart.service.ts:
```diff
@@ -1,40 +1,88 @@
+
+  getTotalNumber() {
+    let counter = 0;
+    this.dataArray.forEach(elem => {
```
Highly recommended to use map and reduce functions

ARHariri commented on src/services/cart.service.ts:
```diff
@@ -1,40 +1,88 @@
+        thumbnail: el.thumbnail,
+        color: 'temp', // TODO: no such color element is received from server
+        size: el.size,
+        quantity: el.quantity
```
Add count property to this object

imans777 committed:
changed styling of the whole bag page structure
change product sliding to only have one product
complete select-count functionality and styling
other major and minor fixes on the other places

imans777 committed:
update numbers
